bots/vpc_isolate.py

```diff
@@ -0,0 +1,174 @@
+'''
+##vpc-isolate
+What it does: turn off dns resource
+              change network acl to new empty one with deny all
+              add iam policy, to all users in the account, which limits vpc use: ec2 and sg use in the vpc
+
+Usage: AUTO: vpc_isolate
+Limitation: None
+'''
+
+import boto3
+from botocore.exceptions import ClientError
+import json
+
+
+def create_acl(vpc):
+
+    text_output = ''
+    # An array which will contain all vpc subnets
+    association_ids = []
+
+    try:
+        # getting all acls within the vpc, to get the subnet associations from them
+        network_acl_iterator = vpc.network_acls.all()
+
+        # Going through all each acl and checking for associated subnets
+        for acl in network_acl_iterator:
+            associations = acl.associations
+            if associations:
+                # gets all subnets from the acl in an array and adding it to association_ids array
+                association_ids += [association.get('NetworkAclAssociationId') for association in associations]
+                
+        # checking if there any subnets to associate with the new acl
+        if association_ids:

```

Comment:
If the array is empty you wont go into the if statement?

Reply:
yes because if array is empty it means vpc has no subnets. without subnets associations acl is useless, so there is no need in creating new network acl.